src/backend/distributed/executor/partitioned_intermediate_results.c

```diff
@@ -437,23 +461,37 @@ PartitionedResultDestReceiverReceive(TupleTableSlot *slot, DestReceiver *copyDes
 	Datum *columnValues = slot->tts_values;
 	bool *columnNulls = slot->tts_isnull;
 
+	int partitionIndex;
+
 	if (columnNulls[partitionedDest->partitionColumnIndex])
 	{
-		ereport(ERROR, (errcode(ERRCODE_NULL_VALUE_NOT_ALLOWED),
-						errmsg("the partition column value cannot be NULL")));
+		if (partitionedDest->skipNullPartitionColumnValue)
+		{
+			/* skip over this tuple, it is not needed for the join */
+			return true;
+		}
+		else
+		{
+			ereport(ERROR, (errcode(ERRCODE_NULL_VALUE_NOT_ALLOWED),
+							errmsg("the partition column value cannot be NULL")));
+		}
 	}
-
-	Datum partitionColumnValue = columnValues[partitionedDest->partitionColumnIndex];
-	ShardInterval *shardInterval = FindShardInterval(partitionColumnValue,
-													 partitionedDest->shardSearchInfo);
-	if (shardInterval == NULL)
+	else
```

Comment:
what is the reason for the else block?

both branches above break out of the function already.
the extra level of indent actually threw me off completely while reading the change.

if this was the las part of the function I could understand, however, given the function continues outside of the else block (which is only reached when entering this else block) I would suggest to remove it here.

It also prevents the `partitionIndex` declaration and assignment separation.